config/rbac/vspherecsiconfig_editor_role.yaml

```diff
@@ -0,0 +1,24 @@
+# permissions for end users to edit vspherecsiconfigs.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: vspherecsiconfig-editor-role
+rules:
+- apiGroups:
+  - csi.tanzu.vmware.com
+  resources:
+  - vspherecsiconfigs
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - csi.tanzu.vmware.com
+  resources:
+  - vspherecsiconfigs/status
+  verbs:
+  - get
```

Comment:
question: we update the status as well, right ?

Reply:
- These permissions are meant for the user (who creates a vspherecsiconfig object), not the controller code. Status sub-resources should not be modified by a user, just the controller code.

```
Status subresource
The /status subresource is used to split the user-provided specification of a CR instance from the controller-provided status. The main motivation for this is privilege separation:

The user usually should not write status fields.

The controller should not write specification fields.
```